### PR TITLE
[Concepts] Introduces circuit breakers

### PR DESCRIPTION
**Pull request by ignasi35**
References #59

This is a copy/paste version of https://doc.akka.io/docs/akka/current/common/circuitbreaker.html .
We will need to setup a canonical URL (#111) in one of the two pages.

### CODE REVIEW

**Comment by ignasi35** [comment]
> I'm not sure what's the best approach here. Maybe linking to the Akka docs would be good enough right now?

@jtownley here's another example of how introducing the Akka Platform guide will impact the location of already-existing documentation (see discussion on #111). This case, the existing content is part of doc.akka.io and is provided by the `akka-docs/` project in https://github.com/akka/akka/. 

My point is, we will need to approach the `Concepts` (and maybe other sections) of the guide with a higher-level thinking.

**Comment by patriknw** [comment]
> I'm not sure what's the best approach here. Maybe linking to the Akka docs would be good enough right now?

I'm not sure I understand what the problem with including the proposed content as a concept here even though it's the same introduction in the Akka reference documentation. It's a high level description.

The reference docs could be changed and instead link to this concept page, but for the reader's convenience it's good with an intro like that also in reference documentation without having to jump to another place.

File: docs-source/docs/modules/concepts/pages/circuit-breakers.adoc [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+    - Callbacks can be used eg. to collect statistics about all invocations or to react on specific call results like success, failures or timeouts.
+    - Supported callbacks are: *onCallSuccess*, *onCallFailure*, *onCallTimeout*, *onCallBreakerOpen*.
+    - These are executed in the `ExecutionContext` provided.
+   
+The following diagram illustrates the Finite State Machine described above:
+[#circuit-breaker-states]
+.States of a Circuit Breaker
+[caption=""]
+image::concepts/circuit-breaker-states.png[States of a Circuit Breaker]
```
Comment (ennru) [inline docs-source/docs/modules/concepts/pages/circuit-breakers.adoc:70]: Add links to the reference docs?

**Comment by ignasi35** [comment]
I just realised we introduce the 'Circuit Breakers' concept but it is never mentioned or used in the tutorial. This is a smell.

IIUC, of all client code used in the sample app for the tutorial, only the akka-persistence-cassandra code has an explicit circuit breaker that the user may tune.

**Comment by patriknw** [comment]
> Circuit Breakers concept but it is never mentioned or used in the tutorial.

I think that is more of an advanced topic that would fit into a how-to. I created issue https://github.com/akka/akka-platform-guide/issues/143 for that.

**Review by ennru — APPROVED** [review]
LGTM.

**Pull request merged**